services/user.py

```diff
@@ -0,0 +1,54 @@
+from django.contrib.auth import get_user_model
+
+
+def create_user(username: str,
+                password: str,
+                email: str = None,
+                first_name: str = None,
+                last_name: str = None) -> None:
+
+    user = get_user_model().objects.create_user(username=username,
+                                                password=password)
+
+    if email:
+        user.email = email
+
+    if first_name:
+        user.first_name = first_name
+
+    if last_name:
+        user.last_name = last_name
+
+    user.save()
+    return user
+
+
+def get_user(user_id: int) -> get_user_model:
+    return get_user_model().objects.get(id=user_id)
+
+
+def update_user(user_id: int, username: str = None,
+                password: str = None,
+                email: str = None,
+                first_name: str = None,
+                last_name: str = None) -> None:
+
+    user = get_user(user_id=user_id)
+
+    if username:
+        user.username = username
+
+    if password:
+        user.set_password(password)
+
+    if email:
+        user.email = email
+
+    if first_name:
+        user.first_name = first_name
+
+    if last_name:
+        user.last_name = last_name
+
+    user.save()
```

Comment:
The same issue with the `save()` method is in the `update_user` function. You call `save()` method twice. The first time is in the `set_password` method of the User model, the second time is on line 53. You should avoid calling `save()` multiple times. You can set the password after all other fields are set and before the `save()` call.